### PR TITLE
Some additional methods for the ObservableList

### PR DESCRIPTION
**Pull request by hendrikebbers**
[Review on Reviewable](https://reviewable.io/reviews/canoo/dolphin-platform/12)

### CODE REVIEW

**Comment by aalmiray** [comment]
Review status: 0 of 4 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[java-core/src/main/java/com/canoo/dolphin/collections/ObservableList.java, line 95 \[r1\]](https://reviewable.io:443/reviews/canoo/dolphin-platform/12#-K8Zvf_rRhPf4h9ZCBqQ:-K8Zvf_rRhPf4h9ZCBqR:-1347522442) ([raw file](https://github.com/canoo/dolphin-platform/blob/cb89c1bff9a47d10d9c7c47ce0cf7620fe4346bf/java-core/src/main/java/com/canoo/dolphin/collections/ObservableList.java#L95)):*
I think it's worth stating that the new `List` is *not* an `ObservableList`

---

*[java-server/src/test/java/com/canoo/dolphin/server/impl/TestObservableArrayListWriteOperations.java, line 338 \[r1\]](https://reviewable.io:443/reviews/canoo/dolphin-platform/12#-K8Zw1qnMP1sm_n_A00B:-K8Zw1qnMP1sm_n_A00C:-464016500) ([raw file](https://github.com/canoo/dolphin-platform/blob/cb89c1bff9a47d10d9c7c47ce0cf7620fe4346bf/java-server/src/test/java/com/canoo/dolphin/server/impl/TestObservableArrayListWriteOperations.java#L338)):*
as a personal preference I like the approach of using given/when/then labels within tests

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/canoo/dolphin-platform/12)*

**Comment by aalmiray** [comment]
Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/canoo/dolphin-platform/12)*

**Comment by hendrikebbers** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[java-server/src/test/java/com/canoo/dolphin/server/impl/TestObservableArrayListWriteOperations.java, line 338 \[r1\]](https://reviewable.io:443/reviews/canoo/dolphin-platform/12#-K8Zw1qnMP1sm_n_A00B:-K8_-cIjPWDEcsNsDMNt:-74042826) ([raw file](https://github.com/canoo/dolphin-platform/blob/cb89c1bff9a47d10d9c7c47ce0cf7620fe4346bf/java-server/src/test/java/com/canoo/dolphin/server/impl/TestObservableArrayListWriteOperations.java#L338)):*
You mean comments?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/canoo/dolphin-platform/12)*

**Comment by aalmiray** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[java-server/src/test/java/com/canoo/dolphin/server/impl/TestObservableArrayListWriteOperations.java, line 338 \[r1\]](https://reviewable.io:443/reviews/canoo/dolphin-platform/12#-K8Zw1qnMP1sm_n_A00B:-K8_4Ris9x57aWDSq66t:-1274493805) ([raw file](https://github.com/canoo/dolphin-platform/blob/cb89c1bff9a47d10d9c7c47ce0cf7620fe4346bf/java-server/src/test/java/com/canoo/dolphin/server/impl/TestObservableArrayListWriteOperations.java#L338)):*
yes. It can be done with comments (`//given:`) or with a standard label (`given:`)

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/canoo/dolphin-platform/12)*

File: java-core/src/main/java/com/canoo/dolphin/collections/ObservableList.java [hunk 1]
```diff
@@ -68,4 +68,32 @@
      * @return a {@link Subscription} instance that can be used to deregister the listener.
      */
     Subscription onChanged(ListChangeListener<? super E> listener);
+
+    /**
+     * Appends all elements to the end of this list
+     * @param elements the elements that should be added to the list
+     * @return <tt>true</tt> if the list changed as a result of the call
+     */
+    boolean addAll(E... elements);
```
Comment (netopyr) [inline java-core/src/main/java/com/canoo/dolphin/collections/ObservableList.java]: We should probably also add variations of these methods that take a starting index in addition.

File: java-core/src/main/java/com/canoo/dolphin/collections/ObservableList.java [hunk 2]
```diff
@@ -68,4 +68,32 @@
...
+     * @return <tt>true</tt> if the list changed as a result of the call
+     */
+    boolean removeAll(E... elements);
+
+
+    /**
+     * Removes all elements from this list and returns a new list that contains the removed elements.
+     * @return a read only list that contains the removed elements
+     */
+    List<E> clearContent();
```
Comment (netopyr) [inline java-core/src/main/java/com/canoo/dolphin/collections/ObservableList.java]: Do we really need this?

**Comment by netopyr** [comment]
Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[java-client/src/test/java/com/canoo/dolphin/client/impl/TestObservableArrayListWriteOperations.java, line 10 \[r1\]](https://reviewable.io:443/reviews/canoo/dolphin-platform/12#-K8cXbx4ntfHHWLFfBhg:-K8cXbx5J8Oz22MjMq7q:-477398944) ([raw file](https://github.com/canoo/dolphin-platform/blob/cb89c1bff9a47d10d9c7c47ce0cf7620fe4346bf/java-client/src/test/java/com/canoo/dolphin/client/impl/TestObservableArrayListWriteOperations.java#L10)):*
* import

---

*[java-server/src/test/java/com/canoo/dolphin/server/impl/TestObservableArrayListWriteOperations.java, line 10 \[r1\]](https://reviewable.io:443/reviews/canoo/dolphin-platform/12#-K8cXyRCDX8CSPw1cxt1:-K8cXyRCDX8CSPw1cxt2:-477398944) ([raw file](https://github.com/canoo/dolphin-platform/blob/cb89c1bff9a47d10d9c7c47ce0cf7620fe4346bf/java-server/src/test/java/com/canoo/dolphin/server/impl/TestObservableArrayListWriteOperations.java#L10)):*
* import

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/canoo/dolphin-platform/12)*

**Comment by hendrikebbers** [comment]
Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[java-core/src/main/java/com/canoo/dolphin/collections/ObservableList.java, line 77 \[r1\]](https://reviewable.io:443/reviews/canoo/dolphin-platform/12#-K8Zp_cG6cKtb9O1ghR0-r1-77:-K8doQ6WvL7mwVmUXaa0:-1506302210) ([raw file](https://github.com/canoo/dolphin-platform/blob/cb89c1bff9a47d10d9c7c47ce0cf7620fe4346bf/java-core/src/main/java/com/canoo/dolphin/collections/ObservableList.java#L77)):*
sounds like a good idea. I will create an issue for that.

---

*[java-core/src/main/java/com/canoo/dolphin/collections/ObservableList.java, line 98 \[r1\]](https://reviewable.io:443/reviews/canoo/dolphin-platform/12#-K8Zp_cG6cKtb9O1ghR0-r1-98:-K8dpPEgLAIwp5H9OcJj:-31285449) ([raw file](https://github.com/canoo/dolphin-platform/blob/cb89c1bff9a47d10d9c7c47ce0cf7620fe4346bf/java-core/src/main/java/com/canoo/dolphin/collections/ObservableList.java#L98)):*
yes, as long as we don't have a GC. Currently you need to this to clear a list:

List<E> copy = new ArrayList<E>(observableList);
observableList.clear();
beanManager.removeAll(copy);

By using this method you can simply write 
beanManager.removeAll(observableList.clearContent());

When we have the GC the method isn't needed anymore and can be removed.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/canoo/dolphin-platform/12)*

**Comment by hendrikebbers** [comment]
Review status: 2 of 4 files reviewed at latest revision, 5 unresolved discussions.

---

*[java-core/src/main/java/com/canoo/dolphin/collections/ObservableList.java, line 98 \[r1\]](https://reviewable.io:443/reviews/canoo/dolphin-platform/12#-K8Zp_cG6cKtb9O1ghR0-r1-98:-K8dsAD-Hhuy1369CCxy:-1879963285) ([raw file](https://github.com/canoo/dolphin-platform/blob/cb89c1bff9a47d10d9c7c47ce0cf7620fe4346bf/java-core/src/main/java/com/canoo/dolphin/collections/ObservableList.java#L98)):*
Since we want to integrate the GC in near future maybe we can remove this method directly. What do you think

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/canoo/dolphin-platform/12)*

**Comment by netopyr** [comment]
Review status: 2 of 4 files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[java-client/src/test/java/com/canoo/dolphin/client/impl/TestObservableArrayListWriteOperations.java, line 17 \[r2\]](https://reviewable.io:443/reviews/canoo/dolphin-platform/12#-K8dttrGaFMdw5a5CatX:-K8dttrGaFMdw5a5CatY:-1824752979) ([raw file](https://github.com/canoo/dolphin-platform/blob/70ff94f6337da7bf243d85fee5fce659be793cf3/java-client/src/test/java/com/canoo/dolphin/client/impl/TestObservableArrayListWriteOperations.java#L17)):*
*-import

---

*[java-core/src/main/java/com/canoo/dolphin/collections/ObservableList.java, line 98 \[r1\]](https://reviewable.io:443/reviews/canoo/dolphin-platform/12#-K8Zp_cG6cKtb9O1ghR0-r1-98:-K8du0bqTdu6V5SZ3R_n:653407398) ([raw file](https://github.com/canoo/dolphin-platform/blob/cb89c1bff9a47d10d9c7c47ce0cf7620fe4346bf/java-core/src/main/java/com/canoo/dolphin/collections/ObservableList.java#L98)):*
I agree, we should remove it immediately

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/canoo/dolphin-platform/12)*

**Comment by hendrikebbers** [comment]
Review status: 2 of 4 files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[java-client/src/test/java/com/canoo/dolphin/client/impl/TestObservableArrayListWriteOperations.java, line 10 \[r1\]](https://reviewable.io:443/reviews/canoo/dolphin-platform/12#-K8cXbx4ntfHHWLFfBhg:-K8zVg-KkRxYDlDeg5Ev:-499146009) ([raw file](https://github.com/canoo/dolphin-platform/blob/cb89c1bff9a47d10d9c7c47ce0cf7620fe4346bf/java-client/src/test/java/com/canoo/dolphin/client/impl/TestObservableArrayListWriteOperations.java#L10)):*
Done.

---

*[java-client/src/test/java/com/canoo/dolphin/client/impl/TestObservableArrayListWriteOperations.java, line 17 \[r2\]](https://reviewable.io:443/reviews/canoo/dolphin-platform/12#-K8dttrGaFMdw5a5CatX:-K8zVgZafhqL5MJKkiFf:-499146009) ([raw file](https://github.com/canoo/dolphin-platform/blob/70ff94f6337da7bf243d85fee5fce659be793cf3/java-client/src/test/java/com/canoo/dolphin/client/impl/TestObservableArrayListWriteOperations.java#L17)):*
Done.

---

*[java-server/src/test/java/com/canoo/dolphin/server/impl/TestObservableArrayListWriteOperations.java, line 10 \[r1\]](https://reviewable.io:443/reviews/canoo/dolphin-platform/12#-K8cXyRCDX8CSPw1cxt1:-K8zVyQMmFprCpn7fW__:-499146009) ([raw file](https://github.com/canoo/dolphin-platform/blob/cb89c1bff9a47d10d9c7c47ce0cf7620fe4346bf/java-server/src/test/java/com/canoo/dolphin/server/impl/TestObservableArrayListWriteOperations.java#L10)):*
Done.

---

*[java-server/src/test/java/com/canoo/dolphin/server/impl/TestObservableArrayListWriteOperations.java, line 338 \[r1\]](https://reviewable.io:443/reviews/canoo/dolphin-platform/12#-K8Zw1qnMP1sm_n_A00B:-K8zW0hwC258R95Cf-eV:-499146009) ([raw file](https://github.com/canoo/dolphin-platform/blob/cb89c1bff9a47d10d9c7c47ce0cf7620fe4346bf/java-server/src/test/java/com/canoo/dolphin/server/impl/TestObservableArrayListWriteOperations.java#L338)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/canoo/dolphin-platform/12)*

**Comment by hendrikebbers** [comment]
mmmh, the build is working on my system. The error looks like Travis can't access Maven Central. maybe Travis currently has some problems...

**Comment by hendrikebbers** [comment]
I have no idea how I can restart the build at Travis

**Comment by hendrikebbers** [comment]
I think this can be merged

**Comment by netopyr** [comment]
Reviewed 4 of 4 files at r3.
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks broke.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/canoo/dolphin-platform/12)*

**Pull request merged**